weekly-openmeteo/api_error.py

```diff
@@ -0,0 +1,39 @@
+'''This module contains the API error class'''
+
+class APIError(Exception):
+    """Describes an error triggered by a failing API call."""
+
+    def __init__(self, message: str, code: int = 500):
+        """Creates a new APIError instance."""
+        self.code = code
+        self._message = message
+        super().__init__(self.message)
+        self.define_error()
+
+    @property
+    def message(self):
+        '''Getter for message of APIError class'''
+        return self._message
+
+    @message.setter
+    def message(self, new_message):
+        '''Setter for the message property of the APIError class'''
+        self._message = f"{self._message}: {new_message}"
+
+    def define_error(self):
+        '''Returns the associated error message to the status code'''
+        error_messages = {
+            400: "Bad Request: The server could not understand the request.",
```

Comment:
Extensive list of errors what clear descriptions. Will be useful if we run into issues with extraction later in the project.